WorldProcessor/WorldProcessor.Core/Services/WorldIterationService.cs

```diff
@@ -0,0 +1,265 @@
+                }
+            }
+
+            return result;
+        }
+
+        private async Task<Dictionary<IPosition, Tile>> IterateWithMutationMagnitudeRecalculationAsync(
+            World world,
+            IEnumerable<TileConfig> tileConfigs,
+            CancellationToken cancellationToken)
+        {
+            var maximalWorldDimensionValue = world.Dimensions.GetMaximalCoordinate();
+
+            var worldOrderedIterationPath = world.Dimensions
+                .GenerateOrderedSpiralPath(
+                    world.Dimensions.GetZeroPosition(),
+                    maximalWorldDimensionValue)
+                .ToList();
+
+            var result = new Dictionary<IPosition, Tile>();
+            foreach(var key in world.Tiles.Keys)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                result[key] = world.Tiles[key];
+            }
+
+            for (int i = 0; i < worldOrderedIterationPath.Count; i++)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var currentPosition = worldOrderedIterationPath[i];
+
+                if (!world.Tiles.TryGetValue(currentPosition, out var currentTile))
+                {
+                    continue;
+                }
+
+                var currentTileConfig = tileConfigs.First(tileConfig => tileConfig.Id == currentTile.ConfigId);
+
+                var orderedRingOfTiles = Enumerable.Range(1, currentTileConfig.CrowdWeightMultiplierRadius)
+                    .SelectMany(radius => world.Dimensions
+                        .GenerateOrderedRingPath(currentPosition, radius))
+                    .ToList();
+
+                for (int j = 0; j < orderedRingOfTiles.Count; j++)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    if (!result.ContainsKey(orderedRingOfTiles[j]))
+                    {
+                        continue;
+                    }
+
+                    var oldMutationChance = result[orderedRingOfTiles[j]].MutationChance;
+
+                    result[orderedRingOfTiles[j]].MutationChance = 
+                        (int)(oldMutationChance * currentTileConfig.CrowdWeightMultiplier);
+                }
+            }
+
+            return result;
+        }
+
+        private bool IsTileNeedToMutate(Tile tile, IPosition position, TileConfig config, World world)
+        {
+            if (config.MinAge == 0 ||
+                tile.BirthEpoch + config.MinAge >= world.Epoch)
+            {
+                return true;
+            }
+
+            if (config.MaxAge != 0 &&
+                tile.BirthEpoch + config.MaxAge <= world.Epoch)
+            {
+                return true;
+            }
+
+            var mutationCheckRandomValue = _randomValueGenerationService
+                .Generate(
+                    world.Seed,
+                    world.Epoch + 1,
```

Comment:
```suggestion
                    world.Epoch,
```
Давай не заглядывать в будущее на эту итерацию )

Reply:
fixed